### ISSUE TITLE
About Keras Siamese network performance

### ISSUE THREAD

**Issue by Juliachang** [fchollet/keras#5052]
Hello guys,

I am trying to use Siamese network to learn the similarity. 
I tried mnist_siamese_graph.py on [Github](https://github.com/fchollet/keras/blob/master/examples/mnist_siamese_graph.py) . After training, the accuracy on training set and on test set are very low and unstable as well.
Is the performance normal for Siamese network? Or do I misunderstand the low accuracy? 
I am new in Siamese network, and sorry for asking basic questions. if anyone can explain it, I'll very appreciate it!
Thank you very much! :-)

**Comment by kgrm** [comment]
Accuracy is not a particularly meaningful metric given the formulation of the problem and the loss function chosen, if you consider that the optimization goal is to get the siamese network to output 0 for matching pairs and high values for non-matching pairs.

**Comment by bstriner** [comment]
@Juliachang See this issue: https://github.com/fchollet/keras/issues/4980 

> Actually we can get test accuracy of 99.5% by reverting the commit: 7f25a77.

**Comment by Juliachang** [comment]
@bstriner Thank you very much!! Problem fixed! :-)
